### DISCUSSION TITLE
When I run forge test after implementing the anvilConfig I'm getting setup failed error

### DISCUSSION THREAD

**Discussion by Rajiv620019** [Cyfrin/foundry-full-course-cu#530] [Q&A] [answered]
## 2023-08-03T16:10:18.867718Z ERROR forge::runner: setUp failed reason="You have an active broadcast already." contract=0x7fa9385be102ac3eac297483dd6233d62b3e1496

**Comment by 0xusmanf** [comment]
You are starting a broadcast before stopping the already running one. Check your setUp function if it's conflicting with any other running broadcast. If you still don't find the problem share your gitHub repo of the project and add more context to your question.

**Reply by Rajiv620019** [reply]
Hey Usman thanks for the reply, I have tried but can't able to solve it 
```
function setUp() external {
        DeployFundMe deployFundMe = new DeployFundMe();
        fundMe = deployFundMe.run();
    }
```
Here is my repo: https://github.com/Rajiv620019/foundry_fund_me

**Reply by 0xusmanf** [reply]
In your `HelperConfig.s.sol` file at line `42` you are starting the broadcast again instead of stoping it that is why you are getting this error, use `vm.stopBroadcast` and your message will go away.

**Reply by Rajiv620019** [reply]
Oops, I should have double-checked it. It was a silly mistake thanks for helping out 👍Happy learning!!! ✌️

**Comment by shikhar229169** [comment, marked as answer]
The problem is in Helper Config file

```solidity
function getAnvilConfig() public returns (NetworkConfig memory) {
        if (activeNetworkConfig.priceFeed != address(0)) {
            return activeNetworkConfig;
        }

        vm.startBroadcast();

        MockV3Aggregator mockV3Aggregator = new MockV3Aggregator(DECIMALS, INITIAL_ANSWER);
        vm.startBroadcast();

        NetworkConfig memory anvilConfig = NetworkConfig({priceFeed: address(mockV3Aggregator)});
        return anvilConfig;
}
```

You added vm.startBroadcast instead of vm.stopBroadcast.

So the corrected code is:

```solidity
function getAnvilConfig() public returns (NetworkConfig memory) {
        if (activeNetworkConfig.priceFeed != address(0)) {
            return activeNetworkConfig;
        }

        vm.startBroadcast();

        MockV3Aggregator mockV3Aggregator = new MockV3Aggregator(DECIMALS, INITIAL_ANSWER);
        vm.stopBroadcast();

        NetworkConfig memory anvilConfig = NetworkConfig({priceFeed: address(mockV3Aggregator)});
        return anvilConfig;
}
```

**Reply by Rajiv620019** [reply]
Hey Shikhar thanks for helping me out. I should have checked it before. 👍👍

**Reply by shikhar229169** [reply]
Great thing is now you know what to do whenever you face the same error.
